pandas/core/missing.py

```diff
@@ -335,18 +349,18 @@ def _interpolate_scipy_wrapper(
         "polynomial",
     ]
     if method in interp1d_methods:
-        if method == "polynomial":
-            method = order
+        kind = order if method == "polynomial" else method
         terp = interpolate.interp1d(
-            x, y, kind=method, fill_value=fill_value, bounds_error=bounds_error
+            x, y, kind=kind, fill_value=fill_value, bounds_error=bounds_error
         )
         new_y = terp(new_x)
     elif method == "spline":
         # GH #10633, #24014
-        if isna(order) or (order <= 0):
-            raise ValueError(
-                f"order needs to be specified and greater than 0; got order: {order}"
-            )
+        if isna(order):
+            raise ValueError(f"order needs to be specified; got order: {order}")
+        assert isinstance(order, int)
```

Comment:
We should not use asserts to please mypy if there is any risk for it to bubble up to the user (without it being a panda bug). I didn't look in enough detail to be sure here, but it looks like a user can pass eg a float here, and it should not raise an AssertionError. 

xref https://github.com/pandas-dev/pandas/issues/37350

Reply:
I'm guessing this is done because isna won't narrow the type. If order is typed correct, `order is None` should be used instead of isna: order must either be a python integer or None. This eliminates the need for the assert.

Reply:
> We should not use asserts to please mypy if there is any risk for it to bubble up to the user (without it being a panda bug). I didn't look in enough detail to be sure here, but it looks like a user can pass eg a float here, and it should not raise an AssertionError. 
> 
> 
> 
> xref https://github.com/pandas-dev/pandas/issues/37350

Fair point, noted